nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/SplitText.java

```diff
@@ -326,6 +329,27 @@ public void process(InputStream in) throws IOException {
         return splitFlowFiles;
     }
 
+    /**
+     * Will merge the two files. This operation has different semantics then
+     * sessioin.merge. Also, it always expects 2 and no more then 2 flow files,
+     * so vararg signature primarily for simplification.
+     */
+    private FlowFile merge(ProcessSession session, FlowFile... flowFiles) {
+        FlowFile mergedFlowFile = session.create();
```

Comment:
@olegz Excuse me if I wasn't clear enough, but I was mentioning about the relationship between original incoming FlowFile and outgoing FlowFiles those are derived from the original FlowFile.

The 2nd one in the var arg is a FlowFile which is derived from the original and it has provenance to the original file, but if we create another FlowFile here, the lineage gets lost.